python/examples/example_errors.ipynb

```diff
@@ -0,0 +1,977 @@
+    "plot_jacobian(rdata)\n",
+    "\n",
+    "print(f\"largest absolute Jacobian value: {np.max(np.abs(rdata.J)):.3g}\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1f6ec0e3",
+   "metadata": {},
+   "source": [
+    "In this case, the default relative error tolerance may be too high and lead too large absolute errors. \n",
+    "\n",
+    "Let's retry simulation using stricter tolerances:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "600ae826",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set stricter relative error tolerance\n",
+    "amici_solver = amici_model.getSolver()\n",
+    "amici_solver.setRelativeTolerance(amici_solver.getRelativeTolerance() / 100)\n",
+    "\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True,\n",
+    "    solver=amici_solver\n",
+    ")\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])\n",
+    "assert all(rdata.status == amici.AMICI_SUCCESS for rdata in res[RDATAS])\n",
+    "print(\"Simulations finished succesfully.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "616710b6",
+   "metadata": {},
+   "source": [
+    "## `Cvode routine CVode returned a root after reinitialization`\n",
+    "\n",
+    "Let's run a simulation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cec31332",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "petab_problem = benchmark_models_petab.get_problem(\"Weber_BMC2015\")\n",
+    "amici_model = import_petab_problem(petab_problem, verbose=False, force_compile=False)\n",
+    "\n",
+    "np.random.seed(4)\n",
+    "problem_parameters = dict(\n",
+    "    zip(\n",
+    "        petab_problem.x_free_ids,\n",
+    "        petab_problem.sample_parameter_startpoints(n_starts=1)[0],\n",
+    "    )\n",
+    ")\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True\n",
+    ")\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4386603",
+   "metadata": {},
+   "source": [
+    "**What happened?**\n",
+    "\n",
+    "The simulation failed because the initial step-size after an event or heaviside function was too small. The error occured during simulation of condition `model1_data1` after successful preequilibration (`model1_data2`)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "02f2dd5d",
+   "metadata": {},
+   "source": [
+    "**How to address?**\n",
+    "\n",
+    "The error message already suggests a fix for this situation, so let's try increasing the relative tolerance:"
```

Comment:
that's good maybe we should do the same for other errors?

Reply:
If we can make good suggestions, that would for sure be helpful.